### PR TITLE
fix(android): BottomNavigation fragment child already has a parent

### PR DESCRIPTION
**Pull request by NathanWalker**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/DevelopmentWorkflow.md#running-unit-tests-application.

## What is the current behavior?

Android BottomNavigation was super fragile, prone to bugs and race conditions due to poor fragment handling.

## What is the new behavior?

Android BottomNavigation is stable and reliable.

closes https://github.com/NativeScript/NativeScript/issues/8132
closes https://github.com/NativeScript/NativeScript/issues/7901
closes https://github.com/NativeScript/NativeScript/issues/9051
closes https://github.com/NativeScript/NativeScript/issues/8251

### CODE REVIEW

File: packages/core/ui/bottom-navigation/index.android.ts [hunk 1]
```diff
@@ -454,28 +454,42 @@ export class BottomNavigation extends TabNavigationBase {
...
-		if (this._currentTransaction) {
-			this._currentTransaction.commitNowAllowingStateLoss();
-			this._currentTransaction = null;
+	private attachFragment(fragment: androidx.fragment.app.Fragment, id?: number, name?: string): void {
+		const fragmentManager = this._getFragmentManager();
+		if (fragment) {
+			if (fragment.isAdded() || fragment.isRemoving()) {
+				// ignore
+			} else {
+				var fragmentExitTransition = fragment.getExitTransition();
```
Comment (rigor789) [inline packages/core/ui/bottom-navigation/index.android.ts:469]:
```suggestion
				const fragmentExitTransition = fragment.getExitTransition();
```

File: packages/core/ui/bottom-navigation/index.android.ts [hunk 2]
```diff
@@ -555,6 +567,34 @@ export class BottomNavigation extends TabNavigationBase {
...
+	private removeFragment(fragment: androidx.fragment.app.Fragment, fragmentManager?: any) {
+		if (!fragmentManager) {
+			fragmentManager = this._getFragmentManager();
+		}
+		if (fragment) {
+			if (!fragment.isAdded() || fragment.isRemoving()) {
+				// ignore
+				return;
+			} else {
+				var fragmentExitTransition = fragment.getExitTransition();
```
Comment (rigor789) [inline packages/core/ui/bottom-navigation/index.android.ts:579]:
```suggestion
				const fragmentExitTransition = fragment.getExitTransition();
```

File: packages/core/ui/bottom-navigation/index.android.ts [hunk 3]
```diff
@@ -555,6 +567,34 @@ export class BottomNavigation extends TabNavigationBase {
...
+				if (fragmentExitTransition && fragmentExitTransition instanceof org.nativescript.widgets.CustomTransition) {
+					fragmentExitTransition.setResetOnTransitionEnd(true);
+				}
+				if (fragment && fragment.isAdded() && !fragment.isRemoving()) {
+					var pfm = (<any>fragment).getParentFragmentManager ? (<any>fragment).getParentFragmentManager() : null;
```
Comment (rigor789) [inline packages/core/ui/bottom-navigation/index.android.ts:584]:
```suggestion
					const pfm = (<any>fragment).getParentFragmentManager ? (<any>fragment).getParentFragmentManager() : null;
```

**Review by rigor789 — APPROVED** [review]
Looks good to me, though I think we could clean up the nesting a little bit to make it a bit easier to follow/read, and perhaps debug when necessary. 

Left some suggestions to change `var` to `const`, but regardless this looks fine to merge. 🚀

**Comment by itsmerockingagain** [comment]
any fix for ns 6?

**Pull request merged**